WalletWasabi.Fluent/ViewModels/Wallets/Send/TransactionPreviewViewModel.cs

```diff
@@ -189,7 +191,14 @@ private async Task BuildAndUpdateAsync()
 		{
 			UpdateTransaction(CurrentTransactionSummary, newTransaction);
 
-			await PrivacySuggestions.BuildPrivacySuggestionsAsync(_wallet, _info, _destination, newTransaction, _isFixedAmount);
+			// Cancel and dispose old token source.
+			_cancellationTokenSource.Cancel();
+			_cancellationTokenSource.Dispose();
```

Comment:
This is a bit dangerous. Others can use the cancellationtoken meanwhile you are disposing it here. Imagine there are multiple locations where the code checking the token status like:

```
await Task.Delay(1000, token);
...
if (token.IsCancellationRequested)
{
   break;
}
```
The second usage of token might fail because it is already disposed. 
What if you pass the cancellation from the dialog and the timeout handled internally, thus creating another cancellationtoken there? Next PR idea.

Reply:
Yes, It smelled to me, too. I will check out your suggestion.